analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/Extensions/IOperationExtensions.cs

```diff
@@ -30,6 +30,7 @@ operation switch
             {
                 _ when IParameterReferenceOperationWrapper.IsInstance(operation) => IParameterReferenceOperationWrapper.FromOperation(operation).Parameter,
                 _ when ILocalReferenceOperationWrapper.IsInstance(operation) => ILocalReferenceOperationWrapper.FromOperation(operation).Local,
+                _ when IFieldReferenceOperationWrapper.IsInstance(operation) => IFieldReferenceOperationWrapper.FromOperation(operation).Field,
```

Comment:
This is not enough. We need to inspect the parent operation as well to avoid `first.obj` to be tracked. See the ticket description.
Please add UT like this to demonstrate the issue with the rule:
```
    public void Method(Program first, Program second)
    {
        Monitor.Exit(second.obj);
        Monitor.Enter(first.obj); 
    }
```
In this case, the `second.obj` should NOT be tracked. So the FP case should not demonstrate itself. 

I'm using this `Exit`/`Enter` pair to demonstrate the issue, because we cannot use branching (yet) to properly demonstrate it in a clear way.

Reply:
On the same note, please add this UT just to document the behavior.
```
    public void Method(Program arg, bool condition)
    {
        Monitor.Enter(arg.obj);     // FN, because we are not field sensitive
        if(condition)
            Monitor.Exit(arg.obj);
    }
```